arbnode/inbox_tracker.go

```diff
@@ -780,3 +780,74 @@ func (t *InboxTracker) ReorgBatchesTo(count uint64) error {
 	log.Info("InboxTracker", "SequencerBatchCount", count)
 	return t.txStreamer.ReorgToAndEndBatch(dbBatch, prevBatchMeta.MessageCount)
 }
+
+func (t *InboxTracker) ReorgMessagesTo(messageCount arbutil.MessageIndex) error {
```

Comment:
What I liked about ReorgBatchTo init option was that it used existing code (also used by the inbox reader).
I think that rolling a few messages too many should be o.k. in the use-cases I can think about.
How about an init option that you give in a message and it rolls back to start of this batch?

I also think we might want an init option that stops processing inbox after a certain message, so we'll know we have what we need in memory for these cases.

Reply:
I agree that providing a message should just roll back to the last batch before the message, simplifies things

Reply:
I think if we do that the option should just be reorg-to-batch. It's unexpected behavior to delete too many messages, and might interfere with debugging.

Reply:
After chatting with Tsahi, I agree that we should just rename option to `reorg-to-batch`.  There is also a concern that blocks will not be removed if they were already moved into `ancients`, not sure if we can detect that and display warning?